rest/route/github.go

```diff
@@ -102,8 +102,14 @@ func (gh *githubHookApi) Parse(ctx context.Context, r *http.Request) error {
 }
 
 func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
+	hasApp := gh.settings.GetGithubAppAuth() != nil
 	switch event := gh.event.(type) {
 	case *github.PingEvent:
+		// Ignore events from GitHub app if app is not set up.
+		// If the app is set up, ignore events from repo webhooks.
+		if (event.GetInstallation() != nil) != hasApp {
```

Comment:
this logic is a little awkward; if they're both not nil or both nil then we want to continue? We may want to move this all out to a helper function that just explains itself (could pass in the event and re-switch on it above this or something)

Reply:
its logically equivalent to 
if (fromApp && !hasApp) ||  (!fromApp && hasApp) {skip} 
which the comment describes more directly

would it be better to make it more clear?